projects/igniteui-angular/src/lib/data-operations/expressions-tree-util.ts

```diff
@@ -1,7 +1,116 @@
+import { DateTimeUtil } from '../date-common/util/date-time.util';
+import { EntityType, FieldType } from '../grids/common/grid.interface';
+import { GridColumnDataType } from './data-util';
+import { IFilteringOperation, IgxBooleanFilteringOperand, IgxDateFilteringOperand, IgxDateTimeFilteringOperand, IgxFilteringOperand, IgxNumberFilteringOperand, IgxStringFilteringOperand, IgxTimeFilteringOperand } from './filtering-condition';
 import { IFilteringExpression } from './filtering-expression.interface';
-import { IFilteringExpressionsTree } from './filtering-expressions-tree';
+import { IExpressionTree, IFilteringExpressionsTree } from './filtering-expressions-tree';
 
 export class ExpressionsTreeUtil {
+    public static recreateTreeFromEntities(tree: IExpressionTree, entities: EntityType[]): IExpressionTree {
+        const isTree = (entry: IExpressionTree | IFilteringExpression): entry is IExpressionTree => {
+            return 'operator' in entry;
+        }
+
+        const entity = entities.find(e => e.name === tree.entity);
+
+        for (let i = 0; i < tree.filteringOperands.length; i++) {
+            const operand = tree.filteringOperands[i];
+            if (isTree(operand)) {
+                tree.filteringOperands[i] = this.recreateTreeFromEntities(operand, entities);
+            } else {
+                if (operand.searchTree) {
+                    operand.searchTree = this.recreateTreeFromEntities(operand.searchTree, entities);
+                }
+                tree.filteringOperands[i] = this.recreateExpression(operand, entity?.fields);
```

Comment:
What if `entity` does not exist, respectively the fields collection is empty? Do we just silently recreate whatever we can?

Reply:
It would essentially return the expression as is, with a possible addition of `conditionName`. It's not ideal, but we decided against rehydrating based on inferred type.